app/assets/javascripts/events/edit_event_form.js

```diff
@@ -7,16 +7,14 @@ var editEventForm = {
             $('#start_time_tz').setToUserTimeZone(this.timezone());
         }
         if ($('#start_time').length && $('#start_date').length) {
-            this.updateDateAndTimeToUserTimeZone($('#next_date'), $('#start_date'), $('#start_time'));
+            this.updateDateAndTimeToUserTimeZone($('#start_date'), $('#start_time'));
         }
     },
-    updateDateAndTimeToUserTimeZone: function (next_date, start_date, start_time) {
-        var normalized_next_date_time = moment.utc(next_date.val() + " " + start_time.val(), "YYYY-MM-DD hh:mm a");
+    updateDateAndTimeToUserTimeZone: function (start_date, start_time) {
         var normalized_start_date_time = moment.utc(start_date.val() + " " + start_time.val(), "YYYY-MM-DD hh:mm a");
-        var local_next_date_time = normalized_next_date_time.tz(this.timezone());
         var local_start_date_time = normalized_start_date_time.tz(this.timezone());
         start_date.val(local_start_date_time.format("YYYY-MM-DD"));
-        start_time.val(local_next_date_time.format("hh:mm A"));
+        start_time.val(local_start_date_time.format("hh:mm A"));
```

Comment:
This was super confusing. We were showing the time as for the "next_date", but date from the "start_date". If the both dates were in different daylight savings period, it was causing changing of the event time, even if user did not change anything in that field.